vscode-extension/src/test/scripts/runner.ts

```diff
@@ -8,13 +8,33 @@ import * as path from 'path';
 
 import {runTests} from '@vscode/test-electron';
 
+const MAX_TRIES = 3;
+
+/**
+ * Runs the tests via `run` up to 3 times, because downloading vscode is
+ * flaky.
+ */
+async function main() {
+  for (let i = 0; i < MAX_TRIES - 1; i++) {
+    try {
+      await run();
+      return;
+    } catch (err) {
+      console.error('Failed to run tests, retrying...');
+    }
+    // wait a few seconds before retrying
+    await new Promise((resolve) => setTimeout(resolve, 5_000));
+  }
+  await run();
```

Comment:
Is having this line required? It looks like this could be removed and maybe the `-1` could also be removed from the `for`. Then it will attempt to run up to 3 times.

Reply:
Sometimes catching and rethrowing errors messes up error stacks, this way the error that gets thrown is untouched :)